### PR TITLE
util-lib: efi variable access refactorings

### PR DESCRIPTION
**Pull request by poettering**
Nothing major, just some refactorings, splitting up of .c/.h files and such.

### CODE REVIEW

File: src/shared/efi-loader.c [hunk 1]
```diff
@@ -1,537 +1,22 @@
...
-
-        typesafe_qsort(list, count, cmp_uint16);
-
-        *options = TAKE_PTR(list);
-
-        return count;
-}
-
-static int read_usec(const char *variable, usec_t *u) {
+static int read_usec(const char *variable, usec_t *ret) {
```
Comment (yuwata) [inline src/shared/efi-loader.c:13]: I think it is better to such renaming of variables or arguments should be in the previous commit.

That is, the commit splitting a file into two or more should only do that for reviewers.

**Review by yuwata — APPROVED** [review]
LGTM. But one comments.

**Comment by jameshilliard** [comment]
@poettering Think this gets us closer to having a bootctl API(#11221)?

**Comment by yuwata** [comment]
Hm?? mkosi tests are now broken with this?

**Pull request merged**